avocado/core/varianter.py

```diff
@@ -45,9 +45,30 @@ def generate_variant_id(variant):
     :return: String compounded of ordered node names and a hash of all
              values.
     """
+    def get_variant_name(variant):
+        """
+        To get the variant full name string
+
+        :param variant: Avocado test variant (list of TreeNode-like objects)
+        :return: Complete variant name string
+        """
+        full_name = []
+        for node in variant:
+            var_str = []
+            _parent = node
+            while _parent:
+                var_str.append(_parent.name)
+                _parent = _parent.parent
+            try:
+                # Let's drop repeated node names and empty string
+                full_name.extend([x for x in var_str[::-1][1:] if x not in full_name])
+            except IndexError:
+                pass
+        return "-".join(full_name).strip('-')
```

Comment:
I can't think of the reason to do the `strip('-')` here.  Maybe you were assuming that `"-".join(...)` would generate a trailing dash?

Reply:
there was a trailing `-`  due to the empty string in the list in my previous local code, I gotrid of that by [1:], and this strip('-') seems no longer needed, looks like it got carried away, tnx!, will remove it.

Reply:
Hi @sathnaga, I saw a comment from you, but I'm not sure if it intended to address this question of mine.

Reply:
@clebergnu thought explained, looks like confused you rather, simple answer my previous generated a output like below so needed a strip('-'), but fixed that later, but missed to remove strip()

```
Multiplex variants (8)
Variant -run-bonding_mode-round-robin-igmp-igmp_min-packets_per_slave-packets_random-bde9
Variant -run-bonding_mode-round-robin-igmp-igmp_min-packets_per_slave-packets_default-934e
Variant -run-bonding_mode-round-robin-igmp-igmp_max-packets_per_slave-packets_random-749e
Variant -run-bonding_mode-round-robin-igmp-igmp_max-packets_per_slave-packets_default-7d35
```